sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/OptimizeJsonExprs.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.optimizer
+
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.plans.logical.{Aggregate, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+
+/**
+ * Simplify redundant json related expressions.
+ */
+object OptimizeJsonExprs extends Rule[LogicalPlan] {
+  override def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+    case p => p.transformExpressions {
+      case JsonToStructs(_, options1, StructsToJson(options2, child, timeZoneId2), timeZoneId1)
+          if options1 == options2 && timeZoneId1 == timeZoneId2 =>
```

Comment:
And .. I think this `options` comparison is kind of error-prone. JSON has options dedicated for read or write but both expressions can take the same options (although it just becomes no-op).

If there's a write or read only option that affects output, the roundtrip results could vary. For example, suppose we have an option `writeOnlyDatesFromTimestamp` that truncates hours, etc from timestamp, and set it to both `from_json` and `to_json`.

Reply:
Hmm, so my initial commit is limit to two options are both empty, do you think it is safer? Per https://github.com/apache/spark/pull/29828#discussion_r492673131, we can look at the json options later and see whether the same option is symmetrical in read and write.